src/notifications/executeNotificationAction.ts

```diff
@@ -1,23 +1,44 @@
 import openHub from "../hub/openHub";
 import {
-  MessageActions,
+  MessageAction,
+  MessageActionsEnum,
   NOTIFICATIONS_OPEN_QUERY_PARAM,
+  OpenHubWithAction,
   StateType,
 } from "../globals/consts";
 
 export default async function executeNotificationAction(
-  selectedActions: MessageActions[] | undefined
+  selectedActions: MessageAction[] | undefined
 ): Promise<void> {
-  if (selectedActions?.includes(MessageActions.OPEN_HUB)) {
+  if (selectedActions?.includes(MessageActionsEnum.OPEN_HUB)) {
     return openHub(StateType.NOTIFICATION)();
   }
 
-  if (selectedActions?.includes(MessageActions.OPEN_NOTIFICATIONS_IN_HUB)) {
+  if (selectedActions?.includes(MessageActionsEnum.OPEN_NOTIFICATIONS_IN_HUB)) {
     return openHub(
       StateType.NOTIFICATION,
       `/home?${NOTIFICATIONS_OPEN_QUERY_PARAM}`
     )();
   }
 
+  const openHubWithActions = selectedActions
+    ?.map((action) => action as OpenHubWithAction)
+    .filter((action) => action.OpenHubWith)
+    .map(({ OpenHubWith: { path, query_params } }) =>
+      openHub(StateType.NOTIFICATION, buildFullHubPath(path, query_params))()
+    );
+
+  if (openHubWithActions?.length) {
+    return Promise.all(openHubWithActions).then();
```

Comment:
Why the `then`?

Reply:
To change it from returning `Promise<void[]>` to `Promise<void>`